internal/controller/sync.go

```diff
@@ -537,128 +458,127 @@ func (r *Route) buildNextCR(ctx context.Context, route *routev1.Route, revision
 			return nil, err
 		}
 	}
+
 	var serialNumber string
 	if metav1.HasAnnotation(route.ObjectMeta, cmapi.SubjectSerialNumberAnnotationKey) {
 		serialNumber = route.Annotations[cmapi.SubjectSerialNumberAnnotationKey]
 	}
-	csr, err := x509.CreateCertificateRequest(
-		rand.Reader,
-		&x509.CertificateRequest{
-			Version:            0,
-			SignatureAlgorithm: signatureAlgorithm,
-			PublicKeyAlgorithm: publicKeyAlgorithm,
-			Subject: pkix.Name{
-				CommonName:         route.Annotations[cmapi.CommonNameAnnotationKey],
-				Country:            countries,
-				Locality:           localities,
-				Organization:       organizations,
-				OrganizationalUnit: organizationalUnits,
-				PostalCode:         postalCodes,
-				Province:           provinces,
-				SerialNumber:       serialNumber,
-				StreetAddress:      streetAddresses,
-			},
-			EmailAddresses: emailAddresses,
-			DNSNames:       dnsNames,
-			IPAddresses:    ipSans,
-			URIs:           uriSans,
-		},
-		key,
-	)
-	if err != nil {
-		return nil, err
-	}
-	csrPEM := pem.EncodeToMemory(&pem.Block{
-		Type:  "CERTIFICATE REQUEST",
-		Bytes: csr,
-	})
 
-	var issuerName string
-	if metav1.HasAnnotation(route.ObjectMeta, cmapi.IngressIssuerNameAnnotationKey) {
-		issuerName = route.Annotations[cmapi.IngressIssuerNameAnnotationKey]
-	} else {
-		issuerName = route.Annotations[cmapi.IssuerNameAnnotationKey]
+	var revisionHistoryLimit *int32
+	if metav1.HasAnnotation(route.ObjectMeta, cmapi.RevisionHistoryLimitAnnotationKey) {
+		historyLimitRaw := route.Annotations[cmapi.RevisionHistoryLimitAnnotationKey]
+
+		parsedLimit, err := strconv.ParseInt(historyLimitRaw, 10, 32)
+		if err != nil {
+			return nil, fmt.Errorf("invalid revision-history-limit annotation %q on %s/%s", historyLimitRaw, route.Namespace, route.Name)
+		}
+
+		typedLimit := int32(parsedLimit)
+		revisionHistoryLimit = &typedLimit
 	}
 
-	cr := &cmapi.CertificateRequest{
+	secretName := route.Name + "-tls-cert"
+
+	// Build the Certificate resource with the collected information
+	// https://cert-manager.io/docs/usage/certificate/
+	cert := &cmapi.Certificate{
 		ObjectMeta: metav1.ObjectMeta{
 			GenerateName: route.Name + "-",
```

Comment:
I think we should use a deterministic Certificate name so we decrease the chances of creating two Certificates that point to the same Secret.

Reply:
Yeah good shout, I'll change to be deterministic!

Reply:
In the future, users might want to share Certificates between routes (to reduce number of issued certificates), so it can be reused.
Not sure how we might do this (maybe a SecretName annotation).

Reply:
Yeah agreed that could be useful in the future!